### PR TITLE
added missing secrets on statefulset template

### LINKED ISSUE #191: [BUG][opensearch] securityConfigSecret not volumeMount(ed)
**Describe the bug**
There is an option to use a predefined Secret for securityConfig (with the individual files within) at securityConfig.config.securityConfigSecret - setting this with the name of the Secret that contains the configs adds the secret to the list of volumes in the StatefulSet as security-config-complete, but does not add it to the list of VolumeMounts.

**To Reproduce**
1. set up a secret with the securityconfig items (config.yml, internal_users.yml, etc..)
2. assign that secret to securityConfig.config.securityConfigSecret
3. secret will be added to a pod, but not mounted.

**Expected behavior**
Should be mounted at the configured path (ie /usr/share/opensearch/plugins/opensearch-security/securityconfig)

**Chart Name**
opensearch

**Host/Environment (please complete the following information):**
 - Helm Version: 3.7.2
 - Kubernetes Version: 1.21.5

**Additional context**

I think this is a quick and easy fix but I'll leave it up to someone who works with this more to show what I'm missing.

The security-config-complete volume can come from a predefined secret or from a data: {} in the values file, as implemented in the statefulset [here](https://github.com/opensearch-project/helm-charts/blob/63847ee78834063f5f79d2f575c22e6301ea6244/charts/opensearch/templates/statefulset.yaml#L143).

However, only the data: {} option is referred to in the volumeMounts for the statefulset [here](https://github.com/opensearch-project/helm-charts/blob/63847ee78834063f5f79d2f575c22e6301ea6244/charts/opensearch/templates/statefulset.yaml#L368).

```
        {{- if .Values.securityConfig.config.data  }}
        - mountPath: {{ .Values.securityConfig.path }}
          name: security-config-complete
        {{- end }}
```

Unless I'm missing something, this should just be an inclusion of the securityConfigSecret option, like:

```
        {{- if or .Values.securityConfig.config.data .Values.securityConfig.config.securityConfigSecret  }}
        - mountPath: {{ .Values.securityConfig.path }}
          name: security-config-complete
        {{- end }}
```

### PR DESCRIPTION
**Pull request by phijojo**
Signed-off-by: Phijo Jospeh <phijo_jo@yahoo.com>

### Description
The Statefulset template is missing 2 secrets when adding individual secrets.
 
### Issues Resolved
 #191 

### Check List
- [x] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [x] Helm chart version bumped
- [x] Helm chart `CHANGELOG.md` updated to reflect change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/helm-charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: charts/opensearch/Chart.yaml [hunk 1]
```diff
@@ -30,3 +30,5 @@ maintainers:
   - name: peterzhuamazon
   - name: prudhvigodithi
   - name: TheAlgo
+  - name: phijojo
```
Comment (bbarani) [inline charts/opensearch/Chart.yaml:33]: We wont be able to add you as a maintainer without enough nominations. Please refer to at [maintainers.md](https://github.com/opensearch-project/.github/blob/main/MAINTAINERS.md#becoming-a-maintainer) guide to understand the process to become a maintainer.

Comment (phijojo) [reply charts/opensearch/Chart.yaml]: Good to know, thanks for sharing the document. I  have removed my name from the maintainer's list.

File: charts/opensearch/templates/statefulset.yaml [hunk 2]
```diff
@@ -190,6 +190,16 @@ spec:
...
+      {{- if .Values.securityConfig.nodesDnSecret }}
+      - name: nodes-dn
+        secret:
+          secretName: {{ .Values.securityConfig.nodesDnSecret }}
+      {{- end }}
+      {{- if .Values.securityConfig.whitelistSecret }}
+      - name: whitelist
+        secret:
+          secretName: {{ .Values.securityConfig.whitelistSecret }}
+      {{- end }}
```
Comment (DandyDeveloper) [inline charts/opensearch/templates/statefulset.yaml:202]: Shouldn't this be allowlist? https://opensearch.org/docs/latest/security-plugin/configuration/yaml/#allowlistyml

File: charts/opensearch/templates/statefulset.yaml [hunk 3]
```diff
@@ -375,6 +385,16 @@ spec:
           name: tenants
           subPath: tenants.yml
         {{- end }}
+        {{- if .Values.securityConfig.whitelistSecret }}
+        - mountPath: {{ .Values.securityConfig.path }}/whitelist.yml
+          name: whitelist
+          subPath: whitelist.yml
+        {{- end }}
```
Comment (DandyDeveloper) [inline charts/opensearch/templates/statefulset.yaml:392]: Shouldn't this be allowlist? https://opensearch.org/docs/latest/security-plugin/configuration/yaml/#allowlistyml

**Review by DandyDeveloper — CHANGES_REQUESTED** [review]
One of the problems with the whitelist is in 1.X its `whitelist` but in 2.X its now become allowlist. 

We need to cater to both by adding a condition on the version of Opensearch being deployed.

File: charts/opensearch/templates/statefulset.yaml (hunk 2 above)
Comment (phijojo) [reply charts/opensearch/templates/statefulset.yaml:202]: Yes, it should be. 
I have used 2.0.0 image, in which the `securityadmin.sh` doesn't check for allowlist.yml instead it's still using whitelist.yml.So I guess `securityadmin.sh` need a change before we use allowlist.yml.

File: charts/opensearch/templates/statefulset.yaml (hunk 3 above)
Comment (phijojo) [reply charts/opensearch/templates/statefulset.yaml:392]: Yes, it should be.
I have used 2.0.0 image, in which the securityadmin.sh doesn't check for allowlist.yml instead it's still using whitelist.yml.So I guess securityadmin.sh need a change before we use allowlist.yml.

**Comment by prudhvigodithi** [comment]
Hey @DandyDeveloper and @phijojo we have `main` and `1.x` branches that support 2.x and 1.x versions, PR can be raised to main branch that supports allowlist and use whitelist in 1.x branch.
https://github.com/opensearch-project/helm-charts#version-and-branching

**Review by TheAlgo — COMMENTED** [review]
Changes looks good, the only thing I am not sure on the whitelist/allowlist thingy. I am fine with whatever @prudhvigodithi suggests

File: charts/opensearch/CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ---
 ## [Unreleased]
 ### Added
+- Updated StatefulSet with node_dn.yml and whitelist.yml
```
Comment (TheAlgo) [inline charts/opensearch/CHANGELOG.md:10]: Do you mind seeing the PR(s) before on updating the CHANGELOG?

**Review by TheAlgo — CHANGES_REQUESTED** [review]

File: charts/opensearch/CHANGELOG.md (hunk 4 above)
Comment (phijojo) [reply charts/opensearch/CHANGELOG.md:10]: Sure, Do you want me to remove this from the current PR and submit a different one?

Comment (prudhvigodithi) [reply charts/opensearch/CHANGELOG.md:10]: Hey @phijojo please consider this as an example
https://github.com/opensearch-project/helm-charts/pull/250/files#diff-96471865d01149e108175156796718f7584b7d5fae730d0cb5dbd98426bcd801

**Comment by prudhvigodithi** [comment]
@phijojo, please test by just keeping `allowlist` for `main` branch and create a PR with `whitelist` for 1.x branch
https://github.com/opensearch-project/security/blob/main/securityconfig/allowlist.yml
https://github.com/opensearch-project/security/blob/main/securityconfig/whitelist.yml
Thank you 
@peterzhuamazon @DandyDeveloper @TheAlgo @bbarani

**Comment by prudhvigodithi** [comment]
Hey since `whitelist` works for both 2.x and 1.x we can push this to add `whitelist` and ignore `allowlist`, until https://github.com/opensearch-project/security/issues/1879 this gets addressed, I have tested with `whitelist` secret and the chart works fine. Thoughts ? @peterzhuamazon @DandyDeveloper @TheAlgo @bbarani @phijojo

**Comment by peterzhuamazon** [comment]
Hi @phijojo any progress to keep this PR going?
Thanks.

**Comment by bbarani** [comment]
Hi @phijojo @prudhvigodithi  Any updates on this PR?

**Comment by prudhvigodithi** [comment]
Hey @phijojo I have converted this PR to draft, please let us know if this PR required and a valid scenario.
Thank you

**Pull request closed without merging**